src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/FlowAnalysis/SymbolUsageAnalysis/SymbolUsageAnalysis.DataFlowAnalyzer.FlowGraphAnalysisData.cs

```diff
@@ -519,28 +518,33 @@ public override void SetTargetsFromSymbolForDelegate(IOperation write, ISymbol s
                     //      Action y = x;
                     //
                     var targetsBuilder = PooledHashSet<IOperation>.GetInstance();
-                    foreach (var symbolWrite in CurrentBlockAnalysisData.GetCurrentWrites(symbol))
-                    {
-                        if (symbolWrite == null)
+                    CurrentBlockAnalysisData.ForEachCurrentWrite(
+                        symbol,
+                        static (symbolWrite, arg) =>
                         {
-                            continue;
-                        }
+                            if (symbolWrite == null)
+                            {
+                                return true;
+                            }
 
-                        if (!_reachingDelegateCreationTargets.TryGetValue(symbolWrite, out var targetsBuilderForSymbolWrite))
-                        {
-                            // Unable to find delegate creation targets for this symbol write.
-                            // Bail out without setting targets.
-                            targetsBuilder.Free();
-                            return;
-                        }
-                        else
-                        {
-                            foreach (var target in targetsBuilderForSymbolWrite)
+                            if (!arg.self._reachingDelegateCreationTargets.TryGetValue(symbolWrite, out var targetsBuilderForSymbolWrite))
                             {
-                                targetsBuilder.Add(target);
+                                // Unable to find delegate creation targets for this symbol write.
+                                // Bail out without setting targets.
+                                arg.targetsBuilder.Free();
+                                return false;
```

Comment:
❗ this needs to return from the caller as well

Reply:
nit: consider doc'ing in code.

Reply:
➡️ Amended to fix.